core/ecschema-metadata/src/Deserialization/Helper.ts

```diff
@@ -252,13 +262,16 @@ export class SchemaReadHelper<T = unknown> {
   private loadSchemaReferenceSync(ref: SchemaReferenceProps): void {
     const schemaKey = new SchemaKey(ref.name, ECVersion.fromString(ref.version));
     const refSchema = this._context.getSchemaSync(schemaKey, SchemaMatchType.LatestWriteCompatible);
+    if (undefined === this._schema) {
```

Comment:
Maybe it would be cleaner to rewrite this method to take the schema in as a parameter.  As called _schema will always be defined but changing the signature makes the method's contract clear.